src/lib.js

```diff
@@ -0,0 +1,195 @@
+import * as UCAN from "./ucan.js"
+import * as CBOR from "@ipld/dag-cbor"
+import * as RAW from "multiformats/codecs/raw"
+import * as UTF8 from "./utf8.js"
+import * as View from "./view.js"
+import * as Parser from "./parser.js"
+import * as Formatter from "./formatter.js"
+import { sha256 } from "multiformats/hashes/sha2"
+import { CID } from "multiformats/cid"
+
+export * from "./ucan.js"
+import { code } from "./ucan.js"
+
+/** @type {UCAN.Version} */
+export const VERSION = "0.8.1"
+
+export const raw = RAW.code
+
+/**
+ * Encodes given UCAN (in either IPLD or JWT representation) and encodes it into
+ * corresponding bytes representation. UCAN in IPLD representation is encoded as
+ * DAG-CBOR which JWT representation is encoded as raw bytes of JWT string.
+ *
+ * @template {UCAN.Capability} C
+ * @param {UCAN.UCAN<C>} ucan
+ * @returns {UCAN.ByteView<UCAN.UCAN<C>>}
+ */
+export const encode = ucan => {
+  switch (ucan.code) {
+    case code:
+      return CBOR.encode({
+        header: {
+          version: ucan.header.version,
+          algorithm: ucan.header.algorithm,
+        },
+        body: {
+          issuer: ucan.body.issuer,
+          audience: ucan.body.audience,
+          capabilities: ucan.body.capabilities.map(Parser.asCapability),
+          expiration: ucan.body.expiration,
+          proofs: ucan.body.proofs,
+          // leave out optionals unless they are set
+          ...(ucan.body.facts.length > 0 && { facts: ucan.body.facts }),
+          ...(ucan.body.nonce && { nonce: ucan.body.nonce }),
+          ...(ucan.body.notBefore && { notBefore: ucan.body.notBefore }),
+        },
+        signature: ucan.signature,
+      })
+    case raw:
+      return /** @type {Uint8Array} */ (UTF8.encode(ucan.jwt))
+    default:
+      return invalidCode(ucan)
+  }
+}
+
+/**
+ * @param {never} ucan
+ */
+const invalidCode = ({ code: unknown }) => {
+  throw new TypeError(
+    `Provided UCAN has unsupported code: ${unknown}, it must be ${code} for CBOR representation or ${raw} for JWT representation`
+  )
+}
+
+/**
+ * Decodes binary encoded UCAN. It assumes UCAN is in primary IPLD
+ * representation and attempts to decode it with DAG-CBOR, if that
+ * fails it falls back to secondary representation and parses it as
+ * a JWT.
+ *
+ * @template {UCAN.Capability} C
+ * @param {UCAN.ByteView<UCAN.UCAN<C>>} bytes
+ * @returns {UCAN.View<C>}
+ */
+export const decode = bytes => {
```

Comment:
@expede so idea here is that you pass arbitrary block (presumably one that came from `UCAN.encode` or compatible implementation) and it will return an object representation.

Code below first attempts to decode the block via dag-cbor if it fails, it just falls back to parsing it as JWT string. That way UCANs issued by this library are in dag-cbor, but any other valid UCAN still can be parsed.

To disambiguate between two `View` decorator adds `code` discriminant field so that `format` and ` encode` can know in which representation UCAN is.